usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -83,6 +83,194 @@ def agency_autocomplete(self, request):
 
         return Response({"results": results})
 
+    def agency_office_autocomplete(self, request):
```

Comment:
To me it seems like there are a lot of queries going on here for what seems to be pretty denormalized data. For example, for the Toptier Agency section, if you don't do the `DISTINCT` operation in the original query below, you actually already have everything you need to populate the subtier agency and offices section. 
```
toptier_agency_search_text_matches = AgencyOfficeAutocompleteMatview.objects.filter(
        toptier_agency_filter
).order_by(*order_by)
```
Why requery for this same data?

In theory, only one query is necessary to get everything you need for the whole response if you match the search text against all possible matching fields, but I'm not sure about how the performance of this would compare to running one query for each of the three sections. Let's discuss tomorrow.